modules/correlation/id-counter.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2023 OneIdentity LLC.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#ifndef ID_COUNTER_H_INCLUDED
+#define ID_COUNTER_H_INCLUDED
+
+#include "glib.h"
+
+typedef struct _IdCounter
+{
+  guint id;
+  guint ref_count;
+} IdCounter;
+
+IdCounter *id_counter_new(void);
+guint id_counter_get_next_id(IdCounter *self);
+IdCounter *id_counter_acquire(IdCounter *self);
+void id_counter_release(IdCounter *self);
```

Comment:
`acquire()` and `release()` is usually called `ref()` and `unref()` in syslog-ng:

<details>

```
$ git grep -o '^[^ ]*_ref(' | code -

lib/ack-tracker/ack_tracker_factory.c:ack_tracker_factory_ref(
lib/cfg-args.c:cfg_args_ref(
lib/cfg-block-generator.c:cfg_block_generator_ref(
lib/cfg-tree.c:log_expr_node_ref(
lib/control/control-command-thread.c:control_command_thread_ref(
lib/control/control-connection.c:control_connection_ref(
lib/dynamic-window-pool.c:dynamic_window_pool_ref(
lib/eventlog/src/evtctx.c:evt_ctx_ref(
lib/eventlog/src/evtrec.c:evt_rec_ref(
lib/filter/filter-expr.c:filter_expr_ref(
lib/filter/filter-re.c:filter_match_set_template_ref(
lib/gsockaddr.c:g_sockaddr_ref(
lib/healthcheck/healthcheck.c:healthcheck_ref(
lib/logmatcher.c:log_matcher_ref(
lib/logmsg/logmsg.c:log_msg_set_saddr_ref(
lib/logmsg/logmsg.c:log_msg_set_daddr_ref(
lib/logmsg/logmsg.c:log_msg_update_ack_and_ref(
lib/logmsg/logmsg.c:log_msg_ref(
lib/logmsg/nvtable.c:nv_table_ref(
lib/logpipe.c:log_pipe_ref(
lib/logqueue.h:log_queue_ref(
lib/logscheduler.c:log_scheduler_options_set_partition_key_ref(
lib/multi-line/multi-line-pattern.c:multi_line_pattern_ref(
lib/stats/stats-registry.c:_assert_when_internal_or_stores_different_ref(
lib/template/compiler.c:parse_msg_ref(
lib/template/templates.c:log_template_ref(
lib/transport/tls-context.c:tls_context_ref(
lib/transport/tls-verifier.c:tls_verifier_ref(
lib/transport/transport-aux-data.h:log_transport_aux_data_set_peer_addr_ref(
lib/transport/transport-aux-data.h:log_transport_aux_data_set_local_addr_ref(
lib/value-pairs/value-pairs.c:value_pairs_ref(
modules/add-contextual-data/context-info-db.c:context_info_db_ref(
modules/correlation/correlation-context.c:correlation_context_ref(
modules/correlation/correlation.c:correlation_state_ref(
modules/correlation/dbparser.c:log_db_parser_set_program_template_ref(
modules/correlation/pdb-program.c:pdb_program_ref(
modules/correlation/pdb-rule.c:pdb_rule_ref(
modules/java/native/java_machine.c:java_machine_ref(
modules/json/dot-notation.c:_compile_dot_notation_array_ref(
modules/json/dot-notation.c:_compile_dot_notation_member_ref(
modules/kafka/kafka-dest-driver.c:kafka_dd_set_key_ref(
modules/kafka/kafka-dest-driver.c:kafka_dd_set_message_ref(
modules/mqtt/destination/mqtt-destination.c:mqtt_dd_set_message_template_ref(
modules/python/python-options.c:python_option_ref(
modules/redis/redis.c:redis_dd_set_command_ref(
modules/timestamp/rewrite-fix-timezone.c:rewrite_fix_time_zone_set_zone_template_ref(
modules/timestamp/rewrite-set-timezone.c:rewrite_set_time_zone_set_zone_template_ref(
```

</details>

Can you please follow this trend here?